src/Microsoft.AspNetCore.Sockets.Abstractions/DefaultConnectionContext.cs

```diff
@@ -66,9 +68,9 @@ public DefaultConnectionContext(string id, IDuplexPipe transport, IDuplexPipe ap
 
         public ClaimsPrincipal User { get; set; }
 
-        public IDictionary<object, object> Metadata { get; set; } = new ConnectionMetadata();
+        public override IDictionary<object, object> Items { get; set; } = new ConnectionMetadata();
```

Comment:
Still need `ConnectionMetadata`?

Reply:
That's correct. The base class does nothing.

Reply:
I see `ConnectionMetadata` makes it safe to get a value that doesn't exist.

```
// Replace the indexer with one that returns null for missing values
object IDictionary<object, object>.this[object key]
{
    get
    {
        if (Items.TryGetValue(key, out var value))
        {
            return value;
        }
        return null;
    }
    set { Items[key] = value; }
}
```

If we're doing it here, should this be the default value of `Items` in Kestrel?

At the very least I don't think we need to override Items on DefaultConnectionContext.

Reply:
That sounds like a fine change to make but right now it's abstract in kestrel.

Reply:
Ah ok, I didn't realize it was an abstract class. Carry on then.

Reply:
You're talking about making it non virtual right?

Reply:
No, I just didn't realize `ConnectionContext` was abstract.